game-app/game-core/src/main/java/games/strategy/triplea/ui/SimpleUnitPanel.java

```diff
@@ -120,7 +125,13 @@ public void setUnitsFromRepairRuleMap(
    */
   public void setUnitsFromCategories(final Collection<UnitCategory> categories) {
     removeAll();
-    for (final UnitCategory category : categories) {
+    List<UnitCategory> unitCategories = new ArrayList(categories);
+    if (unitCategories.isEmpty()) {
```

Comment:
minor, seems like we can check if `categories` is empty first & return sooner (ie: this line can be moved up by one, and check `categories` instead of `unitCategories`

Reply:
Agree, I create a fix.